File: src/main/java/com/ftn/service/MeasurementUnitService.java
package com.ftn.service;

import com.ftn.model.MeasurementUnit;

import java.util.List;

/**
 * Created by Alex on 5/20/17.
 */
public interface MeasurementUnitService {

    List<MeasurementUnit> read();

    MeasurementUnit read(Long id);

    MeasurementUnit create(MeasurementUnit measurementUnit);

    MeasurementUnit update(Long id, MeasurementUnit measurementUnit);

    void delete(Long id);
}

File: src/main/webapp/service/measurementUnitService.js
/**
 * Created by Alex on 5/21/17.
 */
app.service('measurementUnitService', function($http){
    return {
        read: function(onSuccess, onError){
            $http.get('/api/measurementUnits').then(onSuccess, onError);
        },
        create: function(measurementUnit, onSuccess, onError){
            $http.post('/api/measurementUnits', measurementUnit).then(onSuccess, onError);
        },
        update: function(measurementUnit, onSuccess, onError){
            $http.patch('/api/measurementUnits/' + measurementUnit.id, measurementUnit).then(onSuccess, onError);
        },
        delete: function (id, onSuccess, onError) {
            $http.delete('/api/measurementUnits/' + id).then(onSuccess, onError);
        }
    }
});